### ISSUE TITLE
Groovy config scripts can execute any code, not just RuleSet configuration

### ISSUE THREAD

**Issue by mkobit** [CodeNarc/CodeNarc#164]
Imagine a `codenarc.groovy` RuleSet config file like the following:

```groovy
ruleset {
  AssertWithinFinallyBlock
}
System.exit(15)
```

Would it make sense to add in restrictions to sandbox the script (possibly something from this [Cédric Champeau's blog](https://melix.github.io/blog/2015/03/sandboxing.html))?

**Comment by chrismair** [comment]
Good point, and Cedric's blog post is interesting. But it doesn't seem like there is a way to reliably and comprehensively prevent what should be prevented while allowing what should be allowed -- i.e., coming up with the correct whitelist or blacklist. The reference to using Java security policy in the comments was interesting as well; perhaps that might be feasible.

Chris

**Comment by Bijnagte** [comment]
Maybe the parsing of the ruleset file can be sandboxed using http://groovy-sandbox.kohsuke.org/

**Comment by dckc** [comment]
I'm interested in developing style rules for object capability security as in [Joe-E][]. It would at least provide a way to audit the source of config files.

[Joe-E]: https://en.wikipedia.org/wiki/Joe-E
